plugins/modules/ec2_vol.py

```diff
@@ -826,6 +826,8 @@ def main():
         if volume:
             volume, changed = update_volume(module, ec2_conn, volume)
             if name:
+                if not tags:
+                    tags = {}
```

Comment:
We need to make sure that if tags was `None`, then purge_tags is set to `False`.  Otherwise this will always wipe existing tags when no tags are passed

Reply:
@tremble I have retrived the existing tags and set them in the tags dict, instead of setting it to an empty dict. Please let me know your views.

Reply:
What happens if purge_tags is True?

Reply:
Hi @alinabuzachis , I have removed this line in the latest fix. AFAIK, when purge_tags = true and tags is not passed to the module call or tags = existing tags, the existing tags will remain unaffected [(Reference).](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/ec2.py#L245) In the current fix (in this PR), instead of setting the tags to {}, I have set it to the existing tags, in the above condition.